class-components/src/components/BottomSection/BottomSection.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+import { useSearchParams, useNavigate, Outlet } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { useGetPeopleQuery } from '../../store/apiSlice';
+import { RootState, AppDispatch } from '../../store/store';
+import {
+  setSelectedItemId,
+  setSelectedItem,
+  setItems,
+  toggleSelectedItem,
+} from '../../store/searchSlice';
+import Flyout from '../Floyt/Flyout';
+import { People } from './BottomSection.type';
+import Pagination from '../Pagination/Pagination';
+import './BottomSection.css';
+
+const BottomSection: React.FC = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const searchTerm = useSelector((state: RootState) => state.search.searchTerm);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const selectedItems = useSelector((state: RootState) => state.search.selectedItems);
+
+  const currentPage = Number(searchParams.get('page')) || 1;
+  const { data, error, isLoading } = useGetPeopleQuery({ search: searchTerm, page: currentPage });
+  const [selectedItem, setSelectedItemLocal] = useState<People | null>(null);
+
+  useEffect(() => {
+    if (data) {
+      dispatch(setItems(data.results));
+    }
+  }, [data, dispatch]);
+
+  useEffect(() => {
+    const detailsId = searchParams.get('details');
+    if (detailsId && data) {
+      const foundItem = data.results.find((item) => item.url.includes(detailsId));
+      setSelectedItemLocal(foundItem || null);
+      dispatch(setSelectedItem(foundItem || null));
+    }
+  }, [searchParams, data, dispatch]);
+
+  const handlePageChange = (page: number) => {
+    const params = new URLSearchParams(searchParams.toString());
+    params.set('page', String(page));
+    setSearchParams(params);
+  };
```

Comment:
I think it would also work without using `new URLSearchParams` since `setSearchParams` accepts an object - basically like this:
```
const handlePageChange = (page: number) => {
    setSearchParams({
      page: String(page),
    });
  };
```